GetIntoTeachingApi/Services/ICrmService.cs

```diff
@@ -32,6 +32,7 @@ public interface ICrmService
         void DeleteLink(Entity source, Relationship relationship, Entity target, OrganizationServiceContext context);
         IEnumerable<Entity> RelatedEntities(Entity entity, string relationshipName, string logicalName);
         Entity MappableEntity(string entityName, Guid? id, OrganizationServiceContext context);
+        Entity NewEntity(string entityName, OrganizationServiceContext context);

```

Comment:
Can we make the interface more explicit and have `NewEntity` and `BlankExistingEntity` as methods here, instead of retaining the `MappableEntity` which in some cases will also call `NewEntity`?